code/modules/mob/dead/observer/observer.dm

```diff
@@ -170,29 +170,26 @@ Works together with spawning an observer, noted above.
 		C.images += target.hud_list[SPECIALROLE_HUD]
 	return 1
 
-/mob/proc/ghostize(var/can_reenter_corpse = 1)
+/mob/proc/ghostize(var/can_reenter_corpse = 1, var/should_set_timer = 1)
 	if(ckey)
 		var/mob/dead/observer/ghost = new(src)	//Transfer safety to observer spawning proc.
 		ghost.can_reenter_corpse = can_reenter_corpse
 		ghost.timeofdeath = src.stat == DEAD ? src.timeofdeath : world.time
 
-
-		//This is duplicated for robustness in cases where death might not be called.
-		//It is also set in the mob/death proc
-		if (isanimal(src))
-			set_death_time(ANIMAL, world.time)
-		else if (ispAI(src) || isDrone(src))
-			set_death_time(MINISYNTH, world.time)
-		else
-			set_death_time(CREW, world.time)//Crew is the fallback
-
+		if(should_set_timer)
+			//This is duplicated for robustness in cases where death might not be called.		
+			//It is also set in the mob/death proc		
+			if (isanimal(src))		
+				set_death_time(ANIMAL, world.time)		
+			else if (ispAI(src) || isDrone(src))		
+				set_death_time(MINISYNTH, world.time)		
+			else		
+				set_death_time(CREW, world.time)//Crew is the fallback

```

Comment:
This approach sets a new var. By default it always passes the check, setting the timer properly. The only moment right now I override that is when you ghost from a dead body. In that case "ghost.initialise_postkey()" takes care of setting up timers first time and skips it for any other time you ghost after that. Now this seems to work properly. I am just not sure if it is worth setting up new var in this proc only to use it once. But I could not think of anything better.